extensions/xcode/src/shared/check-path-or-increment.ts

```diff
@@ -0,0 +1,23 @@
+import { joinPathComponents } from "../shared/join-path-components";
+import * as os from "os";
+import { existsAsync } from "../shared/fs-async";
+
+/**
+ * return a valid file path
+ * @param location: location in which we want to save the playground file.
+ * @param filename: filename of the created or opened playground file.
+ * @param forceCreate: define if we want to force create or not.
+ */
+export async function getValidPath(location: string, filename: string, forceCreate: boolean): Promise<string> {
```

Comment:
I am not really sure if this function would be better placed in the `XcodeSwiftPlaygroundService` class 🤔

The implementation of the `getValidPath` function is tightly coupled to creating a Swift Playground based on the `.playground` file extension and therefore not really meant to be shared across the codebase of the Xcode Raycast Extension itself.